mu-plugins/central-hub/tests/phpunit/integration/meta-data/getCustomFieldsValues.php

```diff
@@ -0,0 +1,220 @@
+
+		$expected_custom_fields = [
+			'event-date' => '10-04-2019',
+			'event-time' => '19:30:00',
+			'venue-name' => 'Carnegie Hall',
+		];
+
+		$this->assertSame( $expected_custom_fields, get_custom_fields_values( $this->post, 'events', $config ) );
+
+		// Clean up database.
+		delete_post_meta( $this->post, 'event-date' );
+		delete_post_meta( $this->post, 'event-time' );
+		delete_post_meta( $this->post, 'venue-name' );
+	}
+
+	/**
+	 * Test get_custom_fields_values() should return custom fields default values when post meta is not in database.
+	 */
+	public function test_should_return_custom_fields_default_values_when_post_meta_is_not_in_database() {
+		$config = [
+			'custom_fields' => [
+				'event-date' => [
+					'is_single' => true,
+					'default'   => '',
+				],
+				'event-time' => [
+					'is_single' => true,
+					'default'   => '',
+				],
+				'venue-name' => [
+					'is_single' => true,
+					'default'   => '',
+				],
+			],
+		];
+
+		// Check that database does not contain post meta.
+		$this->assertSame( '', get_post_meta( $this->post, 'event-date', true ) );
+		$this->assertSame( '', get_post_meta( $this->post, 'event-time', true ) );
+		$this->assertSame( '', get_post_meta( $this->post, 'venue-name', true ) );
+
+		$expected_custom_fields = [
+			'event-date' => '',
+			'event-time' => '',
+			'venue-name' => '',
+		];
+
+		$this->assertSame( $expected_custom_fields, get_custom_fields_values( $this->post, 'events', $config ) );
+	}
+
+	/**
+	 * Test get_custom_fields_values() should register a custom callback and return true with callback priority level.
+	 */
+	public function test_filter_registers_custom_callback_and_returns_true_with_cb_priority_level() {
+		$config = [
+			'custom_fields' => [
+				'event-date' => [
+					'is_single' => true,
+					'default'   => '',
+				],
+				'event-time' => [
+					'is_single' => true,
+					'default'   => '',
+				],
+				'venue-name' => [
+					'is_single' => true,
+					'default'   => '',
+				],
+			],
+		];
+
+		// Add post meta to the database so we have something to call.
+		add_post_meta( $this->post, 'event-date', '10-12-2019' );
+		add_post_meta( $this->post, 'event-time', '19:30:00' );
+		add_post_meta( $this->post, 'venue-name', 'Carnegie Hall' );
+
+		// Check database that post meta was added.
+		$this->assertSame( '10-12-2019', get_post_meta( $this->post, 'event-date', true ) );
+		$this->assertSame( '19:30:00', get_post_meta( $this->post, 'event-time', true ) );
+		$this->assertSame( 'Carnegie Hall', get_post_meta( $this->post, 'venue-name', true ) );
+
+		// Register custom callback to filter $tag with priority of 20 and 3 available arguments.
+		add_filter( 'filter_meta_box_custom_fields',  __NAMESPACE__ . '\filter_custom_field_values', 20, 3 );
+		/*
+		 * Filter the custom field values with a custom callback
+		 */
+		function filter_custom_field_values( $custom_fields ) {
+
```

Comment:
@rgadon107 You want to test the following within the callback:

1. The callback receives only 3 arguments.
2. The values for each is as expected.

For assertion 1, use `func_get_args`. Why? You want to grab _all arguments passed to the callback_ to ensure only 3 are received.

Doesn’t defining the function params do this? No. Why? A function can receive _more_ arguments than it specifies in its params list (but it can’t receive less). Look at your code above. You’ve registered that the callback will receive 3 args, but you only defined 1 param, named `$custom_fields`. Although you only defined 1 params for the function, it will actually (should) receive 3.

Does that make sense?

Reply:
@hellofromtonya 
>Does that make sense?

Yes. I updated the custom callback to pass in the 3 callback parameters. The filter passes in each one of them to the callback when the number of args is set to 3. I ran `func_get_args()` inside the cb and used `var_dump()` to check the value. The initial (unfiltered) values returned to `get_custom_field_values()` were all there. 

>For assertion 1, use func_get_args. Why? You want to grab all arguments passed to the callback to ensure only 3 are received.

I'm not sure how to write a valid assertion to test that the 3 unfiltered args were passed to the cb. There's a couple of reasons why this is problematic. 

1) PHPUnit will not allow me to run `$this` inside a non-object context (e.g. the custom callback registered to the filter). 

2) If I return `func_get_args()` to the custom cb, and then invoke the cb as an expected value within the assertion, PHPUnit throws a fatal error. The custom cb is also not a callable; I can't invoke the function name in a string as an expected value and get the assertion to run. 

=============================================

## Doesn't work

	function filter_custom_field_values( $custom_fields, $meta_box_id, $post_id ) {
			$args = func_get_args();
                        
                        $this->assertSame( $args, get_custom_field_values( $this->post, 'events', $config )
		}

----------------------------------------------------------------

	function filter_custom_field_values( $custom_fields, $meta_box_id, $post_id ) {
			return func_get_args();
		}

 $this->assertSame( filter_custom_field_values( $custom_fields, $meta_box_id, $post_id ),
			get_custom_fields_values( $post_id, 'events', $config ) );

----------------------------------------------------------------

	function filter_custom_field_values( $custom_fields, $meta_box_id, $post_id ) {
			return func_get_args();
		}

 $this->assertSame( 'filter_custom_field_values',
			get_custom_fields_values( $post_id, 'events', $config ) );

Reply:
@rgadon107 Why can’t the embedded function access `$this`? It is not a function of the object. Therefore, it does not have access to the object.

As I noted elsewhere, I’m not a fan of functions within a function (ie embedded functions). Why? Readability. Difficulty understanding scope, ie what is available to the embedded function.

I suggest moving the embedded function/callback to be a public method of the test class. Then `$this` is available. Why? By moving the function to a method, the callback is a method of the test object.